### DISCUSSION TITLE
Conselho: Javascript com Nodejs no back-end ?

### DISCUSSION THREAD

**Discussion by KevinRicci** [frontendbr/forum#1454] [Perguntas e Respostas]
Olá pessoal, esta é uma dúvida sobre a carreira no mundo da programação... Estudei Java por um tempo e depois de pesquisar migrei meus estudos para Javascript, por questões de mercado e porquê gostei muito da linguagem depois de perder o "medo". Altamente flexível comparada com Java. Mas fico um tanto pensativo com a ligação íntima de Javascript com Front-End, no qual eu não sou muito chegado(gosto de back). Gostaria de saber se realmente há a possibilidade de trabalhar com Javascript majoritariamente no Back-End, não como full-stack que engloba os dois mundos mas sim programar Javascript com NodesJS no lado servidor. As empresas estão realmente fazendo seus back-end's em javascript ? 

E já emendando, há plena possibilidade de aliar Javascript com práticas e ferramentas DevOps e Cloud Computing ? Acredito que não haja restrições quanto a isso mas é sempre bom saber melhor.

Obrigado.

**Comment by fdaciuk** [comment]
Oi @KevinRicci! Alguns devs ainda associam JS especificamente ao frontend mais por conta da API DOM - que nada tem a ver diretamente com a linguagem, é uma API para navegadores web escrita em JS - do que da linguagem em si.

Mas é perfeitamente possível - e bastante usado hoje em dia - usar JavaScript do lado do servidor sim, não apenas como backend de aplicações, como também para tooling :)

Vale muito a pena estudar a fundo a linguagem e ficar ligado nas novidades que todo ano são lançadas, para conseguir aproveitar ao máximo o que o JS tem a oferecer 😊

**Comment by mihazs** [comment]
Depende muito da sua aplicação.
Eu tenho a preferência por usar PHP nos meus backends, principalmente pelo fato de existirem ferramentas mais maduras para lidar com banco de dados relacionais (mysql) e pela performance muito superior à do node.
Porem, em sistemas que exigem websockets, tarefas assíncronas ou similares, eu optaria por Javascript no backend. Ele lida muito bem com 

Apesar de ter aprendido programar em Java, não recomendo pra ninguém fazer o backend em Java, o código fica muito complexo e amarrado. Você perde muito tempo configurando.

A vantagem de fazer em Javascript é que você trabalha com uma única linguagem pra stack inteira.

**Comment by gabriel-hahn** [comment]
Vejo bastante vagas para NodeJS no mercado, principalmente em empresas que lidam diretamente com sistemas de uma estrutura baseada em I/O. Porém nada comparado a Java ou PHP por exemplo em relação a quantidade de vagas disponíveis.
Em relação ao DevOps, tenho visto que o NodeJS possui um ambiente relativamente maduro, além das características de alta escalabilidade da plataforma.

**Comment by praganavictor** [comment]
No meu caso procurnado vagas Jr esse mes eu fiz umas entrevistas para backend em node que eu achei o pessoal bem exigente quanto a padrão de codigo, testes e tudo que o back envolve. Mas existe mano você consegue Node é muito legal de trabalhar.

**Comment by dittrichlucas** [comment]
Não sei em que ponto a sua dúvida ainda persiste, mas eu acredito ser plenamente possível atuar com JS no back-end. A empresa em que eu trabalhava está construindo um SaaS com o back-end todo em TS. Acredito, inclusive, que vc deve ter encontrado muitas vagas para back em NodeJS. Caso não, a comunidade pode te ajudar mandando algumas vagas (recomendo, caso não tenha acessado ainda, participar das comunidades nodebr no Telegram e Slack).

Quanto a junção com DevOps, nisso não posso te ajudar ainda, mas tenho visto uma compatibilidade muito grande.

Espero ter ajudado!
